Core/ExpressionFunctions.cpp

```diff
@@ -511,23 +511,32 @@ ExpressionValue expFuncReadAscii(const std::wstring& funcName, const std::vector
 
 	file.setPos((long)start);
 
-	unsigned char* buffer = new unsigned char[length];
-	file.read(buffer,(size_t)length);
-
+	unsigned char buffer[1024];
+	bool stringTerminated = false;
 	std::wstring result;
-	for (size_t i = 0; i < (size_t) length; i++)
+
+	for (size_t progress = 0; !stringTerminated && progress < (size_t) length; progress += sizeof(buffer))
 	{
-		if (buffer[i] < 0x20 || buffer[i] > 0x7F)
+		const size_t bytesRead = file.read(buffer, std::min(sizeof(buffer), length - progress));
```

Comment:
This causes a compilation error on MSVC because `sizeof(buffer)` is `size_t` and `length - progress` is `int64_t`. I'd suggest casting `sizeof(buffer)` to `int64_t` as well.

Additionally, you may want to change `progress` to type `int64_t` as well, same as `length`. Currently, this will cause erratic behavior on files that are >4 GB. For example when doing `readascii("file.bin", 0, 0x100000500)` on a `file.bin` that is `0x100000500` bytes long, only `0x500` bytes will be read (or `0x800` after making the above change), rather than the whole file - whereas if `file.bin` is only `0x600` bytes long, then the entire file will be read.

Reply:
Oops, my bad about the typing issues, I can fix that.

There is potentially a much deeper issue though, which is that `BinaryFile.setPos()` takes a `long`, and as far as I can tell, under MSVC, this is 32-bit anyway. I'm not totally sure if this is an issue, and if it is, whether addressing it is in the scope of this PR.